legend-engine-xt-persistence-protocol/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/persistence/test/assertion/status/ActiveRowsEquivalentToJsonAssertFail.java

```diff
@@ -0,0 +1,23 @@
+// Copyright 2022 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.protocol.pure.v1.model.packageableElement.persistence.test.assertion.status;
+
+import org.finos.legend.engine.protocol.pure.v1.model.test.assertion.status.AssertFail;
+
+public class ActiveRowsEquivalentToJsonAssertFail extends AssertFail
```

Comment:
Why we creating new assertFail classes. Just to mention we didn't intend to have assertFail for each assertType and reuse if possible to have better scalability

Reply:
We have had some internal discussion, and here is what we think: as the two new assertion types are essentially different logical entities from the existing EqualToJson, we feel we should not try to reuse or try to establish a relationship with EqualToJson. We shall take it that at the moment it happens to be the case that ActiveRowsEquivalentToJsonAssertFail looks like EqualToJsonAssertFail, but it is entirely possible that it evolves into something different, hence a new class.

Reply:
Can you expand more here?  As Hardik mention, we should avoid creating yet another specific AssertFail subtype. 

These errors are propagated, and need to be consumed on multiple places, potentially including the UI.

We need to come with a way that information is propagated without needing to narrow it down to the actual underlying assertion.

For example, it seems we might be missing a "UnexpectedValueAssertionFail" that can capture expected vs actual.